src/Access/Authentication.h

```diff
@@ -35,6 +36,9 @@ class Authentication
         /// SHA1(SHA1(password)).
         /// This kind of hash is used by the `mysql_native_password` authentication plugin.
         DOUBLE_SHA1_PASSWORD,
+
+        /// Password is checked by a [remote] LDAP server. Connection will be made at each authentication attempt.
+        LDAP_PASSWORD,
```

Comment:
Seems it would be nice to rename the authentication type to just `LDAP` because it has nothing to do with passwords in ClickHouse when you create users. I mean you don't have to specify password when you create an user, e.g.

```
CREATE USER ... IDENTIFIED WITH ldap SERVER ...
```

Reply:
Will do.

Reply:
I would also recommend converting that `enum` to `enum class`.

Reply:
Renamed `LDAP_PASSWORD` enum value to `LDAP_SERVER`.